### ISSUE TITLE
@urql/preact client is seen as emotion cache

### ISSUE THREAD

**Issue by dios-david** [FormidableLabs/urql#933]
I use the following stack:
- preact
- @urql/preact
- preact-ssr-prepass
- theme-ui (a layer on top of emotion)

SSR works fine, but when the app hydrates on the client I see this error:
```
bundle.js:3352 Uncaught TypeError: Cannot read property 'tokvmb' of undefined
    at insertStyles (bundle.js:3352)
```

This error comes from `emotion` when it tries to read it's cache (via an emotion cache provider), but the `cache` object points to a urql client:

<img width="831" alt="Screenshot 2020-08-06 at 16 54 51" src="https://user-images.githubusercontent.com/3816644/89547715-62f75080-d806-11ea-854f-ce54ebdbe9ab.png">

This app works fine with `urql`, the issue is present only when using `@urql/preact`.

**Steps to reproduce**

1. Clone [this branch](https://github.com/dios-david/preact-hydration-repro/tree/urql-preact-with-emotion)
2. Run: `npm i && npm run start`
3. Open http://localhost:3000
4. 💥

**Comment by JoviDeCroock** [comment]
Hey @dios-david

This is not so much an urql issue but I did find the essence of it, razzle doesn't deduplicate well so it essentially creates two namespaces.

because `@urql/preact` imports from `preact` itself (not compat) and compat exports from `preact` this should get deduplicated but when I inspect the bundle I see two instances of `createContext` floating around.

So what I did to fix this is go into the distribution of `@urql/preact` and transform the `preact` import used for `createContext` to `preact/compat` this made it work.

There are a few options for you here:
- use `patch-package` to do the above
- use `urql` itself so you will have two things using `preact/compat`
